docs/CHANGELOG.md

```diff
@@ -0,0 +1,3240 @@
+<!-- \page changelog Changelog -->
```

Comment:
I would remove all the commits before the earliest known release. They are just noise. We can add a note that only the commit history is available for those.

Reply:
I added notes for semantics and previous commits, though they do need some polishing.

Reply:
> I would remove all the commits before the earliest known release. They are just noise. We can add a note that only the commit history is available for those.

So, will we remove this latest one as well, when we get the latest release or will we append the newer release to the list.

Reply:
I was looking at https://github.com/AdarshRawat1/p4c/blob/Docs-Changelogs/CHANGELOG.md

The version that included previous releases was fine, but that list only went until [v1.2.2](https://github.com/p4lang/p4c/releases/tag/v1.2.2). We can keep the changelog until that point, then delegate to the git commit history before that.

Reply:
Oh Thanks !! , I understand it now.